tools/distrib/python/grpc_prefixed/generate.py

```diff
@@ -0,0 +1,138 @@
+# Copyright 2020 The gRPC Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Generates grpc-prefixed packages using template renderer.
+
+To use this script, please use 3.7+ interpreter. This script is work-directory
+agnostic. A quick executable command:
+
+    python3 tools/distrib/python/grpc_prefixed/generate.py
+"""
+
+import dataclasses
+import logging
+import os
+import shutil
+import subprocess
+import sys
+
+import jinja2
+
+WORK_PATH = os.path.realpath(os.path.dirname(__file__))
+LICENSE = os.path.join(WORK_PATH, '../../../../LICENSE')
+BUILD_PATH = os.path.join(WORK_PATH, 'build')
+DIST_PATH = os.path.join(WORK_PATH, 'dist')
+
+env = jinja2.Environment(
+    loader=jinja2.FileSystemLoader(os.path.join(WORK_PATH, 'templates')))
+
+LOGGER = logging.getLogger(__name__)
+
+
+@dataclasses.dataclass
+class PackageMeta:
+    """Meta-info of a PyPI package."""
+    name: str
+    name_long: str
+    destination_package: str
+    version: str = '1.0.0'
+
+
+def clean() -> None:
+    try:
+        shutil.rmtree(BUILD_PATH)
+    except FileNotFoundError:
+        pass
+
+    try:
+        shutil.rmtree(DIST_PATH)
+    except FileNotFoundError:
+        pass
+
+
+def generate_package(meta: PackageMeta) -> None:
+    # Makes package directory
+    package_path = os.path.join(BUILD_PATH, meta.name)
+    os.makedirs(package_path, exist_ok=True)
+
+    # Copy license
+    shutil.copyfile(LICENSE, os.path.join(package_path, 'LICENSE'))
+
+    # Generates source code
+    for template_name in env.list_templates():
+        template = env.get_template(template_name)
+        with open(
+                os.path.join(package_path,
+                             template_name.replace('.template', '')), 'w') as f:
+            f.write(template.render(dataclasses.asdict(meta)))
+
+    # Creates wheel
+    job = subprocess.Popen([
+        sys.executable,
+        os.path.join(package_path, 'setup.py'), 'sdist', '--dist-dir', DIST_PATH
+    ],
+                           cwd=package_path,
+                           stdout=subprocess.PIPE,
```

Comment:
If the subprocess creates too much output, this pipe could fill up and [deadlock the process](https://docs.python.org/3/library/subprocess.html#subprocess.Popen.wait).

Reply:
Good catch. Updated to use `communicate()` instead.

Reply:
I don't think this is the best option either. This will stall the process for 60 seconds and `job.returncode` will be `None`. Maybe use a tempfile or redirect to `sys.stdout` instead?

Reply:
Actually, considering that this script is only going to be run once, it's not that big of a concern. LGTM.